jlm/llvm/ir/types.hpp

```diff
@@ -360,10 +366,19 @@ class StructType::Declaration final
     return *util::AssertedCast<const valuetype>(Types_[index].get());
   }
 
+  [[nodiscard]] std::shared_ptr<const valuetype>
```

Comment:
`const std::shared_ptr<const valuetype>&` ???

Reply:
cannot return reference as there is a cast in the function -- the stored types should be fixed